sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1761,4 +1763,67 @@ private static void checkEmptyStringAllowed(Class<?> type, JavaType genericType,
     }
     return supportedRunners.build();
   }
+
+  /**
+   * @param properties the properties to use to create the pipeline options instance.
+   * @return a pipeline options instance based on the specified properties.
+   */
+  public static PipelineOptions fromProperties(final Properties properties) {
+    if (properties == null) {
+      return create();
+    }
+    final String[] args = properties.stringPropertyNames()
+      .stream()
+      .map(k -> String.format("--%s=%s", k, properties.getProperty(k)))
```

Comment:
Don't use the command line parsing; this isn't a command line. Directly create the PipelineOptions.

Reply:
I can miss something but only args are parsed to this just builds args to ensure we keep a single instantiation mecanism. I don't want to go through a JSON model here but really reuse the plain args parsing to enforce consistency accross the usages.

Reply:
There should be a single shared mechanism between args and properties, yes. It should accept a map or some structured value - I haven't dug in this code for a while. But a fake cmdline is not how to instantiate an object.

Reply:
What is the issue? I have to admit I kind of like it enforces CLI support is not broken and cant be. The alternative makes it very tempting to not respect it.